NebulaWorld/SimulatedWorld.cs

```diff
@@ -524,12 +523,27 @@ public static void RenderPlayerNameTagsOnStarmap(UIStarmap starmap)
                     }
 
                     // Put the marker directly on the location of the player
-                    starmapMarker.rectTransform.anchoredPosition = rectPoint;
-                    starmapMarker.transform.localScale = Vector3.one;
+                    starmapTracker.position = adjustedVector;
+
+                    if (playerModel.Movement.localPlanetId > 0)
+                    {
+                        PlanetData planet = GameMain.galaxy.PlanetById(playerModel.Movement.localPlanetId);
+                        var rotation = planet.runtimeRotation * 
+                            Quaternion.LookRotation(playerModel.PlayerModelTransform.forward, playerModel.Movement.GetLastPosition().LocalPlanetPosition.ToVector3());
+                        starmapTracker.rotation = rotation;
+                    }
+                    else
+                    {
+                        var rotation = Quaternion.LookRotation(playerModel.PlayerModelTransform.forward, playerModel.PlayerTransform.localPosition);
+                        starmapTracker.rotation = rotation;
+                    }
+
+                    starmapTracker.localScale = UIStarmap.isChangingToMilkyWay ? Vector3.zero : 
+                        Vector3.one * (starmap.screenCamera.transform.position - starmapTracker.position).magnitude;
 
                     // Put their name above or below it
-                    nameText.rectTransform.anchoredPosition = new Vector2(rectPoint.x, rectPoint.y + (rectPoint.y >= -350.0 ? -19f : 19f));
-                    nameText.transform.localScale = Vector3.one;
+                    nameText.rectTransform.anchoredPosition = new Vector2(rectPoint.x + (rectPoint.x > 600f ? -35 : 35), rectPoint.y + (rectPoint.y > -350.0 ? -19f : 19f));
```

Comment:
Have you tested this at the minimum resolution? I found some interesting behaviour with it, hence me picking the values I did when I wrote this

Reply:
I changed it to match what the game itself does, so there wouldn't be differing behavior. (although the game also rounds the values, but i didn't implement that.)